src/ucs/debug/debug.h

```diff
@@ -18,6 +19,17 @@ BEGIN_C_DECLS
  */
 void ucs_debug_disable_signal(int signum);
 
+#ifdef __SANITIZE_ADDRESS__
+#define UCS_ASAN_ADDRESS_IS_VALID(_ptr, _size) \
+    do { \
+        char _buffer[_size]; \
+        memcpy(_buffer, _ptr, _size); \
```

Comment:
can we avoid actually doing memcpy?
how about checking __asan_region_is_poisoned for example?
BTW, are we sure some compiler will not optimize-out this memory copy?